subsys/bluetooth/shell/cs.c

```diff
@@ -194,6 +194,184 @@ static int cmd_cs_test_simple(const struct shell *sh, size_t argc, char *argv[])
+		} else if (!strcmp(argv[j], "pbr-both")) {
+			params.main_mode_type = BT_CONN_LE_CS_CONFIG_MAIN_MODE_2;
+			params.sub_mode_type = BT_CONN_LE_CS_CONFIG_SUB_MODE_3;
+		} else if (!strcmp(argv[j], "both-pbr")) {
+			params.main_mode_type = BT_CONN_LE_CS_CONFIG_MAIN_MODE_3;
+			params.sub_mode_type = BT_CONN_LE_CS_CONFIG_SUB_MODE_2;
+		} else if (!strcmp(argv[j], "steps")) {
+			if (++j == argc) {
+				shell_help(sh);
+				return SHELL_CMD_HELP_PRINTED;
+			}
+
+			params.min_main_mode_steps = strtoul(argv[j], NULL, 10);
+			if (++j == argc) {
+				shell_help(sh);
+				return SHELL_CMD_HELP_PRINTED;
+			}
+
+			params.max_main_mode_steps = strtoul(argv[j], NULL, 10);
+			if (++j == argc) {
+				shell_help(sh);
+				return SHELL_CMD_HELP_PRINTED;
+			}
+
+			params.mode_0_steps = strtoul(argv[j], NULL, 10);
+		} else if (!strcmp(argv[j], "aa-only")) {
+			params.rtt_type = BT_CS_TEST_RTT_AA_ONLY;
+		} else if (!strcmp(argv[j], "32b-sound")) {
+			params.rtt_type = BT_CS_TEST_RTT_32_BIT_SOUNDING;
+		} else if (!strcmp(argv[j], "96b-sound")) {
+			params.rtt_type = BT_CS_TEST_RTT_96_BIT_SOUNDING;
+		} else if (!strcmp(argv[j], "32b-rand")) {
+			params.rtt_type = BT_CS_TEST_RTT_32_BIT_RANDOM;
+		} else if (!strcmp(argv[j], "64b-rand")) {
+			params.rtt_type = BT_CS_TEST_RTT_64_BIT_RANDOM;
+		} else if (!strcmp(argv[j], "96b-rand")) {
+			params.rtt_type = BT_CS_TEST_RTT_96_BIT_RANDOM;
+		} else if (!strcmp(argv[j], "128b-rand")) {
+			params.rtt_type = BT_CS_TEST_RTT_128_BIT_RANDOM;
+		} else if (!strcmp(argv[j], "phy-1m")) {
+			params.cs_sync_phy = BT_CS_TEST_CS_SYNC_LE_1M_PHY;
+		} else if (!strcmp(argv[j], "phy-2m")) {
+			params.cs_sync_phy = BT_CS_TEST_CS_SYNC_LE_2M_PHY;
+		} else if (!strcmp(argv[j], "phy-2m-2b")) {
+			params.cs_sync_phy = BT_CS_TEST_CS_SYNC_LE_2M_2BT_PHY;
+		} else if (!strcmp(argv[j], "chmap-rep")) {
+			if (++j == argc) {
+				shell_help(sh);
+				return SHELL_CMD_HELP_PRINTED;
+			}
+
+			params.channel_map_repetition = strtoul(argv[j], NULL, 10);
+		} else if (!strcmp(argv[j], "hat-shape")) {
+			params.ch3c_shape = BT_CONN_LE_CS_CONFIG_CH3C_SHAPE_HAT;
+		} else if (!strcmp(argv[j], "x-shape")) {
+			params.ch3c_shape = BT_CONN_LE_CS_CONFIG_CH3C_SHAPE_X;
+		} else if (!strcmp(argv[j], "chsel-3b")) {
+			params.channel_selection_type = BT_CONN_LE_CS_CONFIG_CHSEL_TYPE_3B;
+		} else if (!strcmp(argv[j], "chsel-3c")) {
+			params.channel_selection_type = BT_CONN_LE_CS_CONFIG_CHSEL_TYPE_3C;
+		} else if (!strcmp(argv[j], "ch3c-jump")) {
+			if (++j == argc) {
+				shell_help(sh);
+				return SHELL_CMD_HELP_PRINTED;
+			}
+
+			params.ch3c_jump = strtoul(argv[j], NULL, 10);
+		} else if (!strcmp(argv[j], "chmap")) {
+			if (++j == argc) {
+				shell_help(sh);
+				return SHELL_CMD_HELP_PRINTED;
+			}
+
+			if (hex2bin(argv[j], strlen(argv[j]), params.channel_map, 10) == 0) {
+				shell_error(sh, "Invalid channel map");
+				return -ENOEXEC;
+			}
+
+			sys_mem_swap(params.channel_map, 10);
```

Comment:
swap ?

Reply:
hex2bin generates big endian value so we're swapping here to make it le